packages/jobs/lib/integration.service.ts

```diff
@@ -32,14 +27,13 @@ class IntegrationService implements IntegrationServiceInterface {
         }
 
         const { runner, activityLogId } = scriptObject;
+        this.runningScripts.set(syncId, { ...scriptObject, cancelled: true });
 
         const res = await runner.client.cancel.mutate({
             syncId
         });
 
-        if (res.isOk()) {
-            this.runningScripts.set(syncId, { ...scriptObject, cancelled: true });
-        } else {
+        if (res.isErr()) {
```

Comment:
fixing a race condition where the runner script could return before the `runningScripts` was updated, causing the `runScript` to return a non cancellation error